Assets/Scripts/Ability/Combat/LaserBeam.cs

```diff
@@ -0,0 +1,61 @@
+using Cysharp.Threading.Tasks;
+using RogueApeStudio.Crusader.HealthSystem;
+using RogueApeStudio.Crusader.Units;
+using System;
+using System.Threading;
+using UnityEngine;
+
+namespace RogueApeStudio.Crusader.Ability.Combat
+{
+    [CreateAssetMenu(fileName = "LaserBeam", menuName = "Ability/Sphere Ability/Laser Beam")]
+    internal class LaserBeam : SphereAbility
+    {
+        [SerializeField] private float _duration;
+        [SerializeField] private float _damageRate;
+        private bool _active;
+        private CancellationTokenSource _cancellationTokenSource;
+
+        private void OnEnable()
+        {
+            _cancellationTokenSource = new();
+        }
+
+        private void OnDisable()
+        {
+            _cancellationTokenSource.Cancel();
+            _cancellationTokenSource?.Dispose();
```

Comment:
This **WILL** always be instaniated, or you've done something awful.
Just give this comment a thumbs up for read.